tanner/tests/test_rfi_emulation.py

```diff
@@ -19,11 +19,12 @@ def test_http_download_fail(self):
         path = 'http://foobarfvfd'
         filename = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNone(filename)
-
+        
     def test_ftp_download(self):
+        self.handler.download_file_ftp = mock.MagicMock()
         path = 'ftp://mirror.yandex.ru/archlinux/lastupdate'
         data = self.loop.run_until_complete(self.handler.download_file(path))
-        self.assertIsNotNone(data)
+        self.handler.download_file_ftp.assert_called_with(yarl.URL(path))
```

Comment:
just for information - we can also write the test which allows to check if we get proper result from the `download_file` function when using ftp.